lib/api/core/auth/tokenManager.js

```diff
@@ -22,38 +43,43 @@ const TIMEOUT_MAX = Math.pow(2, 31) - 1;
  * When a token is invalidated, it cleans up the corresponding connection's subscriptions, and notify the
  * user that the token has expired
  *
+ * @class TokenManager
  * @param {Kuzzle} kuzzle
- * @constructor
  */
-function TokenManager (kuzzle) {
-  /*
-   * Tokens are sorted by their expiration date
-   *
-   * The token id is added to the key to handle
-   * equality between different tokens sharing
-   * the exact same expiration date
-   */
-  this.tokens = new SortedArray([], (a, b) => {
-    if (a.idx === b.idx) {
-      return 0;
-    }
+class TokenManager {
+  constructor(kuzzle) {
+    /** @type Kuzzle */
+    this.kuzzle = kuzzle;
+
+    /*
+     * Tokens are sorted by their expiration date
+     *
+     * The token id is added to the key to handle
+     * equality between different tokens sharing
+     * the exact same expiration date
+     */
+    this.tokens = new SortedArray([], (a, b) => {
+      if (a.idx === b.idx) {
+        return 0;
+      }
 
-    return a.idx < b.idx ? -1 : 1;
-  });
+      return a.idx < b.idx ? -1 : 1;
+    });
 
-  this.timer = null;
+    this.timer = null;
+  }
 
-  this.runTimer = () => {
+  runTimer() {
     if (this.tokens.array.length > 0) {
-      let delay = Math.min(this.tokens.array[0].expiresAt - Date.now(), TIMEOUT_MAX);
+      const delay = Math.min(this.tokens.array[0].expiresAt - Date.now(), TIMEOUT_MAX);
 
       if (this.timer) {
         clearTimeout(this.timer);
       }
 
-      this.timer = setTimeout(this.checkTokensValidity, delay);
+      this.timer = setTimeout(this.checkTokensValidity.bind(this), delay);
```

Comment:
Oh yes, I did not notice the `setTimeout` !! (probable my glasses are dirty... :/ )